include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -993,6 +993,38 @@ AffineConstraints<number>::close()
          ExcMessage("The constraints represented by this object have a cycle. "
                     "This is not allowed."));
 
+  // Now also compute which indices we need in distribute().
+  //
+  // This processor owns only part of the vectors we will work on. one may think
+  // that every processor should be able to simply communicate those elements it
+  // owns and for which it knows that they act as sources to constrained DoFs to
+  // the owner of these DoFs. This would lead to a scheme where all we need to
+  // do is to add some local elements to (possibly non-local) ones and then call
+  // compress().
+  //
+  // Alas, this scheme does not work as evidenced by the disaster of bug
+  // #51, see http://code.google.com/p/dealii/issues/detail?id=51 and the
```

Comment:
this link doesn't exist anymore.

Reply:
Yes, I don't know what to do with this. I moved this code from another place. I replaced the link by text saying that the history of this bug was lost in a transition.